src/app/more/MoreList.model.ts

```diff
@@ -0,0 +1,34 @@
+import { InfiniteData, useInfiniteQuery } from "@tanstack/react-query";
+import { Dummy } from "./MoreList.types";
+import { getListRecommends } from "./MoreList.controller";
+import { useInView } from "react-intersection-observer";
+
+const useMoreListModel = () => {
+  const infinityQuery = useInfiniteQuery<
+    Dummy[],
+    Object,
+    InfiniteData<Dummy[]>,
+    [_1: string, _2: string],
+    number
+  >({
+    queryKey: ["moreList", "recommends"],
+    queryFn: getListRecommends,
+    initialPageParam: 1,
+    getNextPageParam: (lastPage, allPage) => (allPage.length * 5) / 5 + 1,

```

Comment:
[ P3 ] 현재 `lastPage` 값은 사용안되고 있는것 같습니다!

```suggestion
    getNextPageParam: (allPage) => (allPage.length * 5) / 5 + 1,

```

Reply:
두 번째 파라미터 인자를 사용하기 위해 임의로 데이터를 추가해놨는데 `_` 처리해놓도록 하겠습니다 !
임시 데이터이기에 해당 로직은 추후에 변경될 것 같습니다 😊